django/applications/catmaid/control/nat/r.py

```diff
@@ -444,23 +444,23 @@ def compute_scoring_matrix(project_id, user_id, matching_sample,
         match_subset = robjects.NULL
         if matching_sample.subset:
             # Find all possible pairs in each subset
-            pairs = []
+            pairs: List[Tuple[Tuple[int, str], Tuple[int, str]]] = []
```

Comment:
Spelling out the types really helps making clear how complex some objects actually are.